### PR TITLE
[search] Use HasName feature instead of ErrorsMade and NameScores for  categorial requests.

### PR DESCRIPTION
**Pull request by tatiana-yan**

### CODE REVIEW

File: search/ranking_info.hpp [hunk 1]
```diff
@@ -48,6 +48,12 @@ struct RankingInfo
   // tokens are categorial ones.
   bool m_falseCats = false;
 
+  // True iff request is categorial.
```
Comment (mpimenov) [inline search/ranking_info.hpp]: s/request/the request/

File: search/ranking_info.hpp [hunk 2]
```diff
@@ -48,6 +48,12 @@ struct RankingInfo
...
+  bool m_categorialRequest = false;
+
+  // True iff feature has name.
```
Comment (mpimenov) [inline search/ranking_info.hpp]: iff the feature has a name

Comment (tatiana-yan) [reply search/ranking_info.hpp]: done

File: search/ranking_info.hpp (hunk 1 above)
Comment (tatiana-yan) [reply search/ranking_info.hpp]: done

**Comment by tatiana-yan** [comment]
@mpimenov PTAL

**Pull request merged**